coderd/schedule/autostop_test.go

```diff
@@ -364,6 +374,115 @@ func TestCalculateAutoStop(t *testing.T) {
 			// expectedDeadline is copied from expectedMaxDeadline.
 			expectedMaxDeadline: dstOutQuietHoursExpectedTime,
 		},
+		{
+			// A user expects this workspace to be online from 9am -> 9pm.
+			// So if a deadline is going to land in the middle of this range,
+			// we should bump it to the end.
+			// This is already done on `ActivityBumpWorkspace`, but that requires
+			// activity on the workspace.
+			name: "AutostopCrossAutostartBorder",
+			// Starting at 9:45pm, with the autostart at 9am.
+			now:                   pastDateNight,
+			templateAllowAutostop: false,
+			templateDefaultTTL:    time.Hour * 12,
+			workspaceTTL:          time.Hour * 12,
+			// At 9am every morning
+			wsAutostart: "CRON_TZ=America/Chicago 0 9 * * *",
+
+			// No quiet hours
+			templateAutoStart: schedule.TemplateAutostartRequirement{
+				// Just allow all days of the week
+				DaysOfWeek: 0b01111111,
+			},
+			templateAutostopRequirement: schedule.TemplateAutostopRequirement{},
+			userQuietHoursSchedule:      "",
+
+			expectedDeadline:    time.Date(pastDateNight.Year(), pastDateNight.Month(), pastDateNight.Day()+1, 21, 0, 0, 0, chicago),
```

Comment:
could you use `pastDateNight.AddDate()` here?

Reply:
`AddDate` cannot add hours. So it would be `AddDate(0,0,1).Add(time.Hour*X)` But the number of hours is 23.25 hours, which is such a strange amount.

`pastDateNight` = 9:45pm on Wednesday
`expectedDeadline` = 9pm Thursday, which is 12hrs + autostart (which is a cron string, not a Go `time.Time`).

It was easier for me to reason because the offset being 23.25 hours is not really the assertion I am making. I want to assert it's 12hrs after autostart.

Reply:
Fair!